packages/app-extensions/src/actions/modules/actionHandlers/legacyAction.js

```diff
@@ -1,30 +1,74 @@
-export default function* (definition, selection, parent, params, config) {
-  const actionDefinition = new window.nice2.netui.actions.model.ClientActionDefinition(definition.id, definition.id)
-  actionDefinition.setEnabled(!definition.readOnly)
-
-  const entityExplorer = new window.nice2.modules.NewClientLegacyActionsModule({
-    entityName: selection.entityName,
-    formName: selection.entityName
-  })
-  entityExplorer.init()
+const loadScript = src => new Promise((resolve, reject) => {
+  const s = document.createElement('script')
+  s.src = src
+  s.onload = resolve
+  s.onerror = reject
+  document.head.appendChild(s)
+})
 
-  const ctx = new window.nice2.modules.entityexplorer.EntityExplorerActionContext(entityExplorer, null)
+const loadCss = src => new Promise((resolve, reject) => {
+  const head = document.getElementsByTagName('head')[0]
+  const link = document.createElement('link')
+  link.rel = 'stylesheet'
+  link.type = 'text/css'
+  link.href = src
+  link.media = 'all'
+  link.onload = resolve
+  link.onerror = reject
+  head.appendChild(link)
+})
 
-  const situation = {
-    entityName: selection.entityName
+const loadSequentially = sources => {
+  const next = sources.shift()
+  if (next) {
+    return next.handler(next.src).then(() => loadSequentially(sources))
   }
+  return Promise.resolve()
+}
 
-  const action = window.NetuiActionRegistry.newActionFromDefinition(actionDefinition, situation, ctx)
+const sources = [
+  {src: '/nice2/javascript/lang.release.js', handler: loadScript},
+  {src: '/nice2/javascript/nice2-ext-newclient-actions.debug.js', handler: loadScript},
+  {src: '/nice2/javascript/nice2-admin.debug.js', handler: loadScript},
+  {src: '/nice2/dwr-all.js', handler: loadScript},
+  {src: '/static/nice_legacy_actions_setup.js', handler: loadScript},
+  {src: '/css/themes/blue-medium.css', handler: loadCss},
+  {src: '/css/nice2-admin.css', handler: loadCss}
+]
 
-  action.getSelectionNumber = function() {
-    return selection.ids.length
-  }
-  action.getSelection = function() {
-    return {
+const initLegacyActionsEnv = () => window.legacyActionsEnvInitialized === true
```

Comment:
Have you noticed that this is a generator? Usually there is no need for Promises and suchlike. And since neither loadScript nor loadCss is async i dont get the hassle. Or am i missing something?